### PR TITLE
add some developer documentation

### PR DESCRIPTION
**Pull request by oliver-sanders**
Add some docs to explain how data is requested/provisioned.

Add some small readme's to explain different aspects of the project structure.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: src/components/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Components
+
+These are reusable components which may be used throughout the app.
+
+* `graphqlFormGenerator` contains components used for auto-generating forms
+  from GraphQL mutations.
+* `cylc` contains components for Cylc things.
+* `core` consider this deprecated, we should have a tidy one day.
```
Comment (MetRonnie) [inline src/components/README.md:8]: Kind of suggests the Alert component inside is deprecated, which it isn't
```suggestion
* `core` - no point in having this separate dir, we should have a tidy one day.
```

File: src/lang/README.md
Comment (MetRonnie) [inline src/lang/README.md:1]: Should be `src/layouts` not `src/lang` ?

File: src/views/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Components
+
+You may want to take the generic part of a view and turn it into a reusable
+component.
+
+E.G. the Tree *view* displays Cylc cycles/tasks/jobs in a collapsible hierarchy.
+It uses the Tree *component* which provides the generic tree logic e.g.
+indentation, expand/collapse, etc. This Tree *component* is also used by
+the Workflows *view*.
```
Comment (MetRonnie) [inline src/views/README.md:27]: To avoid confusion with the Workspace view which is a bit niche anyway.
```suggestion
the GScan component.
```

Comment (oliver-sanders) [reply src/views/README.md:27]: I don't agree with this one.

Comment (MetRonnie) [reply src/views/README.md:27]: It's technically more correct, but take it or leave it

https://github.com/cylc/cylc-ui/blob/f6b2b31fcd0f7546eccbd78e26281888bce9ee8e/src/views/Workflows.vue#L18-L20

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**